assignment/poker/PokerEvaluator.md

```diff
@@ -0,0 +1,167 @@
+# Poker Hand Strength Evaluator
+
+## Task Description
+
+Your task is to develop an algorithm for comparing the strength of poker hands. 
+
+### Poker Games
+
+You should support one to three poker game types (the more the better):
+* Texas Hold'em - input is board of 5 cards, and multiple hands of 2 cards each
+
+  A value of a Texas Hold'em hand is the best possible value out of all possible subsets of 
+  5 cards from the 7 cards which are formed by 5 board cards and 2 hand cards.
+  See [Texas Hold'em rules](https://en.wikipedia.org/wiki/Texas_hold_%27em).
+
+* Omaha Hold'em - input is board of 5 cards, and multiple hands of 4 cards each
+
+  A value of an Omaha Hold'em hand is the best possible value out of all possible subsets of 
+  5 cards from the 7 cards which are formed by 3 out of 5 board cards and 2 out of 4 hand cards. 
+  See [Omaha Hold'em rules](https://en.wikipedia.org/wiki/Omaha_hold_%27em).
+
+* Five Card Draw - input is multiple hands of 5 cards each
+
+  A value of a Five Card Draw is the value of the 5 hand cards.
+  See [Five Card Draw rules](https://en.wikipedia.org/wiki/Five-card_draw).
+
+### Hand Value
+
+The hand values (for 5 cards) in descending order (from strongest to weakest):
+
+* `Straight Flush` - a `Straight` (see below) which is also a `Flush` (see below)
+* `Four of a kind` - Four cards of the same value
+* `Full House` - Combination of `Three of a kind` and a `Pair`
+* `Flush` - 5 cards of the same suit
+* `Straight` - Sequence of 5 cards in increasing value (note an exception - `A` can both precede `2` and follow `K`)
+* `Three of a kind` - Three cards with the same value
+* `Two pairs` - Two `Pair`-s
+* `Pair` - two cards of the same value
+* `High card` - the "fallback" in case no other hand value rule applies
+
+In case of ties the ranks of the cards forming the combinations decide the highest value. 
+In case of further ties, the ranks of remaining cards (in the order of the rank) decide the highest value. 
+All suits are considered equal in strength.
+
+When comparing `Full House`-s, the `Three of a kind` rank comparison is more important than the `Pair` rank 
+comparison, for example, `QQQ88 > 999KK`, `KKK77 > QQQJJ` and `KKK77 > KKK66`.
+
+When comparing `Straight`-s, the `A2345` `Straight` is the weakest one and the `TJQKA` one the strongest one,
+for example, `23456 > A2345` and `TJQKA > 9TJQK`.
+
+If any of these rules are not clear, please consult [Poker hand and categories](https://en.wikipedia.org/wiki/List_of_poker_hands#Hand-ranking_categories) and
+the game rules of the specific poker games.
+
+## Implementation
+
+### Input and Output
+
+Your code should read from standard input (`stdin`) and write to standard output (`stdout`) until `EOF` is 
+reached (see [standard streams](https://en.wikipedia.org/wiki/Standard_streams)).
+
+The program will be tested by piping input to `stdin` and piping output to `stdout`, with up to 100,000 
```

Comment:
'Piping output to' sounds confusing, 'reading/taking output from'?

Reply:
Also mb give a snippet to make this even more clear :)
```
cat input.txt | your_app > output.txt
```

Reply:
Changed phrasing.

Reply:
test.sh has the latter example.

Reply:
Did the phrasing change? I am new to GitHub :)

Reply:
It did change, but I clarified it yet again. I hope it's better now.